File: android-native/app/src/main/java/com/example/kpkn/screens/workout/WorkoutReadinessBridge.kt
package com.example.kpkn.screens.workout

/**
 * Temporary in-memory bridge to pass readiness adjustments from
 * ReadinessGateScreen to WorkoutScreen during the navigation transition.
 */
object WorkoutReadinessBridge {
    data class ReadinessAdjustments(
        val neural: Int?,
        val muscular: Int?,
        val spinal: Int?,
        val perMuscle: Map<String, Int>,
        /** sleepQuality (1-5) del DailyWellbeingLog más reciente para el sleep modifier. */
        val sleepQuality: Int? = null,
    )

    private var pending: ReadinessAdjustments? = null

    fun store(adjustments: ReadinessAdjustments) {
        pending = adjustments
    }

    fun consume(): ReadinessAdjustments? {
        val result = pending
        pending = null
        return result
    }
}

File: ios-native/KPKNFit/KPKNFit/Presentation/Screens/Workout/WorkoutReadinessBridge.swift
import Foundation

struct WorkoutReadinessBridge {
    struct ReadinessAdjustments {
        let neural: Int?
        let muscular: Int?
        let spinal: Int?
        let perMuscle: [String: Int]
        let sleepQuality: Int?
    }

    private static var pending: ReadinessAdjustments? = nil

    static func store(adjustments: ReadinessAdjustments) {
        pending = adjustments
    }

    static func consume() -> ReadinessAdjustments? {
        let result = pending
        pending = nil
        return result
    }
}
